subprojects/gst-plugins-bad/sys/msdk/gstmsdkdec.c

```diff
@@ -1797,6 +1793,105 @@ gst_msdkdec_create_buffer_pool (GstMsdkDec * thiz, GstVideoInfo * info,
   }
 }
 
+#ifndef _WIN32
+static GstBufferPool *
+gst_msdk_create_va_pool (GstMsdkDec * thiz, GstVideoInfo * info,
+    guint num_buffers)
+{
+  GstBufferPool *pool = NULL;
+  GstAllocator *allocator;
+  GArray *formats = NULL;
+  GstAllocationParams alloc_params = { 0, 31, 0, 0 };
+  GstVaDisplay *display = NULL;
+  GstCaps *caps = NULL;
+
+  display = (GstVaDisplay *) gst_msdk_context_get_va_display (thiz->context);
+
+  if (thiz->use_dmabuf)
+    allocator = gst_va_dmabuf_allocator_new (display);
+  else {
+    formats = g_array_new (FALSE, FALSE, sizeof (GstVideoFormat));
+    g_array_append_val (formats, GST_VIDEO_INFO_FORMAT (info));
+    allocator = gst_va_allocator_new (display, formats);
+  }
+
+  if (!allocator) {
+    GST_ERROR_OBJECT (thiz, "Failed to create allocator");
+    if (formats)
+      g_array_unref (formats);
+    return NULL;
+  }
+
+  caps = gst_video_info_to_caps (info);
+  pool =
+      gst_va_pool_new_with_config (caps,
+      GST_VIDEO_INFO_SIZE (info), num_buffers, num_buffers,
+      VA_SURFACE_ATTRIB_USAGE_HINT_GENERIC, GST_VA_FEATURE_AUTO,
```

Comment:
If it is for decoder, the flag should be VA_SURFACE_ATTRIB_USAGE_HINT_DECODER, I think.

The Encoder seems to have the same problem.

Reply:
> If it is for decoder, the flag should be VA_SURFACE_ATTRIB_USAGE_HINT_DECODER, I think.
> 
> The Encoder seems to have the same problem.

Ok, I will change to the right flag and update the cartwheel patch. Thanks